GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -511,6 +500,35 @@ Brush UpdateCommitButtonAndGetBrush(IReadOnlyList<GitItemStatus> status, bool sh
 
                 return brush;
             }
+
+            void WorkaroundToolbarLocationBug()
+            {
+                // Layout engine bug (?) which may change the order of toolbars
+                // if the 1st one becomes longer than the 2nd toolbar's Location.X
+                // the layout engine will be place the 2nd toolbar first
+
+                // 1. Clear all toolbars
+                toolPanel.TopToolStripPanel.Controls.Clear();
+
+                // 2. Add all the toolbars back in a reverse order, every added toolbar pushing existing ones to the right
+                ToolStrip[] toolStrips = new[] { ToolStripScripts, ToolStripFilters, ToolStripMain };
+                foreach (ToolStrip toolStrip in toolStrips)
+                {
+                    toolPanel.TopToolStripPanel.Controls.Add(toolStrip);
+                }
+
+#if DEBUG
+                // 3. Assert the layout is expected
+                for (int i = toolStrips.Length - 1; i > 0; i--)
+                {
+                    // Assert all toolbars on the same row
+                    Debug.Assert(toolStrips[i].Top == 0, $"{toolStrips[i].Name} must be on placed on the 1st row");
+
+                    // Assert the correct order of toolbars
```

Comment:
```suggestion
                // 3. Assert all toolbars on the same row
                foreach (ToolStrip toolStrip in toolStrips)
                {
                    Debug.Assert(toolStrip.Top == 0, $"{toolStrip.Name} must be placed on the 1st row");
                }

                // 4. Assert the correct order of toolbars
                for (int i = toolStrips.Length - 1; i > 0; i--)
                {
```

Reply:
Why is it better?

Reply:
Because it asserts *all* toolbars as the comment promises.